buildpack.md

```diff
@@ -87,6 +94,11 @@ Buildpack API versions:
  - MUST be in form `<major>.<minor>` or `<major>`, where `<major>` is equivalent to `<major>.0`
  - When `<major>` is greater than `0` increments to `<minor>` SHALL exclusively indicate additive changes
 
+## Types of Buildpacks
+
+- `userspace buildpack` - a traditional buildpack that does not run as root, and has access to the app being built.
```

Comment:
I think we need a different word rather than `userspace`. User space is defined in constrast to kernel space and doesn't refer to the privileges of the user. https://en.wikipedia.org/wiki/User_space

Maybe something generic like `regular buildpack` or `component buildpack`?

Reply:
Changed to `application buildpack` in c3f60ac